src/resdk/resources/background_task.py

```diff
@@ -0,0 +1,72 @@
+"""Process resource."""
+import logging
+from time import sleep, time
+
+from resdk.exceptions import ResolweServerError
+
+from .base import BaseResource
+
+
+class BackgroundTask(BaseResource):
+    """Background task resource.
+
+    :param resolwe: Resolwe instance
+    :type resolwe: Resolwe object
+    :param model_data: Resource model data
+
+    """
+
+    endpoint = "task"
+
+    READ_ONLY_FIELDS = BaseResource.READ_ONLY_FIELDS + (
+        "started",
+        "finished",
+        "status",
+        "description",
+        "output",
+    )
+    WRITABLE_FIELDS = ()
+
+    def __init__(self, resolwe, **model_data):
+        """Initialize attributes."""
+        self.logger = logging.getLogger(__name__)
+
+        #: started
+        self.started = None
+        #: finished
+        self.finished = None
+        #: status - Possible values:
+        #: WA (waiting)
+        #: PR (processing)
+        #: OK (done)
+        #: ER (error)
+        self.status = None
+        #: description
+        self.description = None
+        #: output - JSON field, the actual value depends on the background task
+        self.output = None
+
+        super().__init__(resolwe, **model_data)
+
+    def result(self, timeout=10, refresh_interval=1.0, final_statuses=["OK"]):
```

Comment:
I suggest renaming the method to `wait` and removing all the validation. It will be much easier to integrate websockets once they are supported in the ReSDK. I would also remove the `refresh_interval` as it won't make sense with websockets, and we don't need it now.

Reply:
OK, done.